src/tests/ftest/container/open.py

```diff
@@ -27,106 +27,141 @@
 import traceback
 import uuid
 
-from avocado import main
 from apricot import TestWithServers
+from test_utils import TestPool, TestContainer
+from avocado.core.exceptions import TestFail
 
-from daos_api import DaosPool, DaosContainer, DaosApiError
-
+RESULT_PASS = "PASS"
+RESULT_FAIL = "FAIL"
+RESULT_TO_NUM = {RESULT_PASS: 0, RESULT_FAIL: 1}
 
 class OpenContainerTest(TestWithServers):
     """
-    Tests DAOS container bad create (non existing pool handle, bad uuid)
-    and close.
+    Tests container's open function. Create 2 pool-container pairs and test
+    the following.
+    1. Use valid pool handle and container UUID. Expected to pass
+    container1.open(pool1.handle, container1_uuid)
+    container2.open(pool2.handle, container2_uuid)
+    2. Use the other container's UUID. Expected to fail
+    container1.open(pool1.handle, container2_uuid)
+    container2.open(pool2.handle, container1_uuid)
+    3. Use the other pool's handle. Expected to fail
+    container1.open(pool2.handle, container1_uuid)
+    container2.open(pool1.handle, container2_uuid)
 
     :avocado: recursive
     """
 
     def test_container_open(self):
         """
-        Test basic container bad create.
+        Test container's open function as described above
 
         :avocado: tags=all,container,tiny,full_regression,containeropen
         """
         self.pool = []
         self.container = []
 
-        # common parameters used in pool create
-        mode = self.params.get("mode", '/run/createtests/createmode/')
-        name = self.params.get("setname", '/run/createtests/createset/')
-        size = self.params.get("size", '/run/createtests/createsize/')
-
-        # pool UIDs & GIDs
-        uid = [
-            self.params.get("uid", "/run/createtests/createuid1/"),
-            self.params.get("uid", "/run/createtests/createuid2/"),
-        ]
-        gid = [
-            self.params.get("gid", "/run/createtests/creategid1/"),
-            self.params.get("gid", "/run/createtests/creategid2/"),
-        ]
+        # Get parameters from open.yaml
+        poh_states = self.params.get("poh", '/run/handle_states/*/')
+        uuid_states = self.params.get("uuid", '/run/container_uuid_states/*/')
+
+        # Derive the test case number from the PASS/FAIL-PASS/FAIL combination
+        poh_state_num = RESULT_TO_NUM[poh_states[0]]
+        uuid_state_num = RESULT_TO_NUM[uuid_states[0]]
+        test_case = (poh_state_num << 1) + uuid_state_num
+        if test_case == 3:
+            # Passing in handle2 and UUID2 to open will overwrite the
+            # container's members, so container1 effectively becomes container2
+            # and that wouldn't be a good test, so skip
+            print("Skip the case with container1.open(pool_handle2, " +
+                  "container_uuid2)")
+            return
```

Comment:
Use `self.cancel(<msg>)` to better indicate that the a test was skipped rather than executed successfully.

Reply:
I see. Thanks for letting me know.